packages/loopring_v3/contracts/impl/libexchange/ExchangeBlocks.sol

```diff
@@ -420,6 +427,64 @@ library ExchangeBlocks
         emit BlockCommitted(S.blocks.length - 1, publicDataHash);
     }
 
+    function validateConditionalTransfers(
+        ExchangeData.State storage S,
+        uint32 newBlockIdx,
+        bytes  memory data,
+        bytes  memory auxiliaryData
+        )
+        private
+    {
+        // Read how many conditional transfers are included in the block
+        uint numConditionalTransfers;
+        assembly {
+            numConditionalTransfers := and(mload(add(data, 104)), 0xFFFFFFFF)
+        }
+        require(auxiliaryData.length == numConditionalTransfers * 6, "INVALID_AUXILIARYDATA");
```

Comment:
if the 'conditions' in all requests are not part of the OCDA, will the operator be able to change the conditions here to cheat? For example, change the salt for a much "easier" condition.

Reply:
The condition responsible for the approval of the internal transfer doesn't matter. The only thing that matters is that user approved some transfer to some account for some reason. The salt is only there to make the transfer unique so an identical transfer can be approved in multiple conditional requests. 

Example: tet's say the user has 2 conditional transfers.
- The user allows the transfer of 1 ETH to the operator for a fast withdrawal
- The user allows the transfer of 1 ETH as the fee for calling an on-chain function for the user

The condition of both request is different, but the resulting transfer is the same: 1 ETH needs to be transferred to the operator. The salt is used to make it unique so the uniqueness check when approving the transfer doesn't detect it as a duplicate. But the result is the same, so it doesn't matter what the condition was that resulted in the approval. The result is just that the 1 ETH is allowed to be transferred to the operator.

Reply:
I see your point: conditions with the same `from, to, token, amount` but different salts are actually equal, the operator has the freedom to choose one of them equliavantely.

My concern is that now the **definition of conditions** is basically the tuple of `from, to, token, amount`. It serves fast withdrawal well enough, but may not be as generalized as it can be. I was thinking that **conditions can be anything programmable using solidity**, for example, if *address X has a balance bigger than 10000 Ether, then transfer 0.1 Ether from my offchain account to accountId 123*. -- It doesn't have a matching onchain transfer the order direction.